test/integration/apps-test-cases.yaml

```diff
@@ -0,0 +1,263 @@
+
+  012 - it successfully creates a m2m app with callbacks:
+    command: auth0 apps create --name integration-test-app-m2mapp3 --type m2m --description M2mApp3 --callbacks https://example.com
+    exit-code: 0
+    stdout:
+      contains:
+        - CALLBACKS            https://example.com
+
+  013 - it successfully creates a regular app with callbacks and outputs in json:
+    command: auth0 apps create --name integration-test-app-regapp4 --type regular --description RegApp4 --callbacks https://example.com,https://google.com --json
+    exit-code: 0
+    stdout:
+      json:
+        callbacks: "[https://example.com https://google.com]"
+
+  014 - it successfully creates a regular app with grants:
+    command: auth0 apps create --name integration-test-app-regapp5 --type regular --description RegApp4 --grants credentials,password
+    exit-code: 0
+    stdout:
+      contains:
+        - GRANTS               client_credentials, password
+
+  015 - it successfully creates a spa app with grants and outputs in json:
+    command: auth0 apps create --name integration-test-app-spaapp3 --type spa --description SpaApp3 --grants refresh-token --json
+    exit-code: 0
+    stdout:
+      json:
+        grant_types: "[refresh_token]"
+
+  016 - it successfully creates a native app with grants and outputs in json:
+    command: auth0 apps create --name integration-test-app-nativeapp2 --type native --description NativeApp2 --grants refresh-token,code  --json
+    exit-code: 0
+    stdout:
+      json:
+        grant_types: "[refresh_token authorization_code]"
+
+  017 - it fails to create a m2m app with device code grant:
+    command: auth0 apps create --name integration-test-app-m2mapp4 --type m2m --description M2mApp4 --grants credentials,device-code
+    exit-code: 1
+    stderr:
+      contains:
+        - "Unable to create application"
+
+  018 - it successfully creates a native app with logout urls and outputs in json:
+    command: auth0 apps create --name integration-test-app-regapp6 --type native --description RegularApp --logout-urls https://*.example.com/logout,https://example.com/logout --json
+    exit-code: 0
+    stdout:
+      json:
+        allowed_logout_urls: "[https://*.example.com/logout https://example.com/logout]"
+
+  019 - it successfully creates a native app with origins and outputs in json:
+    command: auth0 apps create --name integration-test-app-regapp7 --type native --description RegularApp --origins https://*.example.com,https://example.com  --json
+    exit-code: 0
+    stdout:
+      json:
+        allowed_origins: "[https://*.example.com https://example.com]"
+
+  020 - it successfully creates a native app with web origins and outputs in json:
+    command: auth0 apps create --name integration-test-app-spaapp4 --type native --description SpaApp4 --web-origins https://example.com  --json
+    exit-code: 0
+    stdout:
+      json:
+        web_origins: "[https://example.com]"
+
+  021 - given a test app:
```

Comment:
Should this be a test case?

Reply:
We need some way to assert that the app used in the below steps is created, otherwise they will all fail. Not the perfect solution unfortunately, as we're quite limited by this commander test runner. Long term I'd love for us to get rid of this way of testing and take https://github.com/cli/cli/blob/trunk/pkg/cmd/issue/list/list_test.go as an example instead.